src/Transport/AzureServiceBusTransport.cs

```diff
@@ -333,5 +333,10 @@ public IWebProxy WebProxy
 
         internal string FullyQualifiedNamespace { get; set; }
         internal TokenCredential TokenCredential { get; set; }
+
+        /// <summary>
+        /// Register call that allows to customize the native ServiceBusMessage just before it is dispatched to the Azure Service Bus SDK client
+        /// </summary>
```

Comment:
How about something like this?

```suggestion
        /// <summary>
        /// Gets or sets the action that allows customization of the native <see cref="ServiceBusMessage"/> 
        /// just before it is dispatched to the Azure Service Bus SDK client. 
        /// </summary>
        /// <remarks>
        /// <para>
        /// This customization is applied after any configured transport customizations, meaning that 
        /// any changes made here may override or conflict with previous transport-level adjustments. 
        /// Exercise caution, as modifying the message at this stage can lead to unintended behavior 
        /// downstream if the message structure or properties are altered in ways that do not align 
        /// with expectations elsewhere in the system.
        /// </para>
        /// </remarks>
```